canonical/ubuntu-advantage-client
# shared MOTD and APT messaging generation

## Proposed Commit Message

shared MOTD and APT messaging generation

APT and MOTD messaging share a number of common messages.
                                                                                
To share those message template between apt-hooks and MOTD-hooks
the ua_update_messaging script will generate message templates
appropriate for both APT and MOTD and store them in
/var/lib/ubuntu-advantage/messages/*tmpl.

The ua_update_messaging script will be triggered on every apt update
and will call esm-apt-hook to render those templates replacing
template variable names with dynamic ESM Apps and Infra package counts
and/or package names.
                                                                                
Fixes: #1536
Fixes: #1561

## Test Steps

build the package from this PR
install on a test system
```bash

sed -i 's/UNRELEASED/xenial'
git commit -am 'changelog'
build-package
sbuild-it ../out/*dsc
lxc launch ubuntu-daily:xenial test-x
lxc file push *deb test-x/
lxc exec test-x bash
dpkg -i /ubuntu-advantage-tools*deb
apt update
run-parts /etc/update-motd.d
ua attach <token>
apt update
run-parts /etc/update-motd.d # to see esm related messages
```

Update messages 
```bash
rm -rf /var/lib/ubuntu-advantage/messages
sudo python3 /usr/lib/ubuntu-advantage/ua_update_messaging.py
for file in /var/lib/ubuntu-advantage/messages/*; do 
  echo ----- $file;
  cat $file;
done

sudo run-parts /etc/update-motd.d
apt upgrade

systemctl list-timers | grep ua-messag
```

## Desired commit type::


## Checklist:
